packages/cdktf/lib/private/fs.ts

```diff
@@ -0,0 +1,72 @@
+import { execSync } from "child_process";
+import * as fs from "fs";
+import * as path from "path";
+import * as os from "os";
+import * as crypto from "crypto";
+
+const HASH_LEN = 32;
+
+// Full implementation at https://github.com/jprichardson/node-fs-extra/blob/master/lib/copy-sync/copy-sync.js
+export function copySync(src: string, dest: string) {
+  function copyItem(p: string) {
+    const sourcePath = path.resolve(src, p);
+    const stat = fs.statSync(sourcePath);
+    if (stat.isFile()) {
+      fs.copyFileSync(sourcePath, path.resolve(dest, p));
+    }
+    if (stat.isDirectory()) {
+      walkSubfolder(p);
+    }
+  }
+
+  function walkSubfolder(p: string) {
+    const sourceDir = path.resolve(src, p);
+    fs.mkdirSync(path.resolve(dest, p), { recursive: true });
+    fs.readdirSync(sourceDir).forEach((item) => copyItem(path.join(p, item)));
+  }
+
+  walkSubfolder(".");
+}
+
+export function archiveSync(src: string, dest: string) {
```

Comment:
Alternatively, we could try to shell out to a nodejs process, so it behaves consistently across all platforms. We would have to bundle something like [archiver](https://www.npmjs.com/package/archiver) as a [jsii dependency](https://aws.github.io/jsii/user-guides/lib-author/typescript-restrictions/) to make this work.

Reply:
I was trying to get around this as it seems less clean 🤔 do we do something similar already somewhere in the project?

Reply:
No, so far we're not doing this. Why do you think it seems less clean?

Reply:
It seems like more moving parts to me, archiver in a bundle, a local node file to execute versus two natively installed binaries seems more execution boundaries. But I don't have a super strong opinion here, we can totally move to archiver if you think it makes more sense

Reply:
> It seems like more moving parts to me, archiver in a bundle, a local node file to execute versus two natively installed binaries seems more execution boundaries.

True, the upside would be that there's no burden on the user to have these binaries installed. e.g. Is `tar` installed by default on Windows? 

Haven't tried shelling out to node in the context of jsii yet. So I'm not sure if that's good or bad.

Reply:
I understand tar is installed by default, but I also agree it could put some burden on the user. I'll move it over to shellout in a single revertable commit, if it causes problems we can fallback to this implementation